portfolio/src/main/webapp/scripts/request_auth.js

```diff
@@ -0,0 +1,188 @@
+import { requestComments } from './request_comments.js';
+import { app } from './firebase.js';
+
+const renderCorrectForm = async () => {
+  const formContainer = document.getElementById('form-container');
+  formContainer.innerHTML = '';
+  firebase.auth().onAuthStateChanged(function(user) {
+    if (user) {
+      formContainer.appendChild(
+        createUserContainer(user.email, user.displayName)
+      );
+      formContainer.appendChild(createCommentForm());
+    } else {
+      formContainer.appendChild(createAuthSelect());
+      formContainer.appendChild(createAuthForm('Login'));
+    }
+  });
+}
+
+renderCorrectForm();
+
+const createUserContainer = (email, username) => {
+  const userContainer = document.createElement('div');
+  userContainer.id = 'user-container';
+  userContainer.className = 'horizontal-container margin-top-bot';
+  
+  const infoContainer = document.createElement('div');
+  const headerText = document.createElement('h2');
+  headerText.innerHTML = 'User';
+  const emailText = document.createElement('p');
+  emailText.innerHTML = `Email: ${email}`;
+  const usernameText = document.createElement('p');
+  usernameText.innerHTML = `Username: ${username}`;
+
+  infoContainer.appendChild(headerText);
+  infoContainer.appendChild(emailText);
+  infoContainer.appendChild(usernameText);
+
+  const logoutButton = createButton('Logout');
+  logoutButton.onclick = function() {
+    app.auth().signOut();
+    renderCorrectForm();
+  }
+
+  userContainer.appendChild(infoContainer);
+  userContainer.appendChild(logoutButton);
+  return userContainer;
+}
+
+const createInput = (value) => {
+  const input = document.createElement('input');
+  const props = {
+    id: value.toLowerCase(),
+    name: value.toLowerCase(),
+    className: 'comment-input',
+    type: value.toLowerCase(),
+    placeholder: value,
+    required: true
+  }
+  Object.assign(input, props);
+  return input;
+}
+
+const handleError = (error) => {
+  alert(error.code);
```

Comment:
Doesn't need to be in this CL but find a more user-friendly way to handle errors.  alerts are just about the most user-hostile client-side pattern this side of the \<blink\> tag.